pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSetMetaData.java

```diff
@@ -233,9 +233,14 @@ private void fetchFieldMetaData() throws SQLException {
 
     StringBuilder sql = new StringBuilder(
         "SELECT c.oid, a.attnum, a.attname, c.relname, n.nspname, "
-            + "a.attnotnull OR (t.typtype = 'd' AND t.typnotnull), "
-            + "pg_catalog.pg_get_expr(d.adbin, d.adrelid) LIKE '%nextval(%' "
-            + "FROM pg_catalog.pg_class c "
+            + "a.attnotnull OR (t.typtype = 'd' AND t.typnotnull), ");
+
+    if ( connection.haveMinimumServerVersion(10)) {
```

Comment:
I guess it should use `ServerVersion` or something around that

Reply:
yes, done